File: BackEnd-Siukat/routes/summary.go
package routes

import (
	"BackEnd-Siukat/services"
	"net/http"

	"github.com/gin-gonic/gin"
)

func SummaryRoutes(r *gin.RouterGroup) {
	group := r.Group("/summary")
	summaryService := services.SummaryService{}

	group.GET("/fakultas", func(c *gin.Context) {
		tahun := c.Query("tahun")
		jalur := c.Query("jalur")
		data, err := summaryService.FetchByFakultas(tahun, jalur)
		if err != nil {
			c.JSON(http.StatusInternalServerError, err)
			return
		}
		c.JSON(http.StatusOK, data)
	})

	group.GET("/prodi", func(c *gin.Context) {
		tahun := c.Query("tahun")
		jalur := c.Query("jalur")
		data, err := summaryService.FetchByProdi(tahun, jalur)
		if err != nil {
			c.JSON(http.StatusInternalServerError, err)
			return
		}
		c.JSON(http.StatusOK, data)
	})
}

File: Be-siukat-new/siukat-service-dev_snmptn_2022/routes/summary.js
var express = require('express');
var router = express.Router();

const Sequelize = require('sequelize');
const Op = Sequelize.Op;
var db = require('../config/database');
var {summarydata} = require("../services");

router.get("/fakultas",function(req,res,next){
    summarydata.fetchByFakultas()
    .then((data)=>{
        res.status(200).json(data)
    }).catch((err)=>{
        res.status(500).json(err)
    })
})
router.get("/prodi",function(req,res,next){
    summarydata.fetchByProdi()
    .then((data)=>{
        res.status(200).json(data)
    }).catch((err)=>{
        res.status(500).json(err)
    }) 
})

var passport = require('passport');

module.exports = router;
